terraform/aurora-mysql/provision/main.tf

```diff
@@ -71,7 +71,7 @@ resource "aws_rds_cluster_instance" "cluster_instances" {
   identifier                            = "${var.instance_name}-${count.index}"
   cluster_identifier                    = aws_rds_cluster.cluster.id
   tags                                  = var.labels
-  instance_class                        = local.serverless ? "db.serverless" : "db.r5.large"
+  instance_class                        = var.instance_class
```

Comment:
**question**: Why have we decided against automatically using `db.serverless`? It seems to be a constant placeholder, and other values don't seem to make sense for serverless in our context. While it is possible to have a cluster with a mix of serverless and non-serverless instances, the cluster will stop being serverless in our case.

Reply:
**suggestion**: Maybe we should document in the serverless_(min|max) properties that they are ineffective without setting the `instance_class` to `db.serverless`?

Reply:
It is a marvellous question, @jhvhs. Thank you very much!

I contemplated the alternative of leaving the hard-coded value there since AWS only accepts a possible instance class for the Serverlessv2 mode. I decided to give the responsibility to the customer and choose the correct instance class because the accepted values for a Serverlessv2 instance class could change in future. The main idea is to modify the code as little as possible.

It is true that with this configuration we won't give the option of having both types of instances, serverless and non-serverless, but it seems very reasonable for the first stage of the service. I would love to offer specific configurations for instances. I guess we will do it in future developments.

I am more pro to documenting the ineffectiveness of serverless_(min|max) properties without setting the instance class to `db.serverless` than hard-coded the value. What do you think?

Reply:
Yes, it feels like documenting is the better way for now

Reply:
Thank you very much!! Done!